cardano-crypto-class/src/Cardano/Crypto/Schnorr.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Crypto.Schnorr (
+  SECP256k1Context,
+  secpContextSignVerify,
+  SECP256k1SchnorrExtraParams,
+  SECP256k1SecKey,
+  SECP256k1SchnorrSig,
+  secpContextCreate,
+  secpContextDestroy,
+  SECP256k1KeyPair,
+  secpKeyPairCreate,
+  secpSchnorrSigSignCustom,
+  SECP256k1XOnlyPubKey,
+  secpKeyPairXOnlyPub,
+  secpSchnorrSigVerify,
+  ) where
+
+import Data.Bits ((.|.))
+import Foreign.Ptr (Ptr)
+import Foreign.C.Types (CUChar, CSize (CSize), CInt (CInt))
+
+data SECP256k1Context
+
+data SECP256k1SchnorrExtraParams
+
+foreign import capi "secp256k1.h secp256k1_context_create"
+  secpContextCreate :: 
+     CInt -- flags
+  -> IO (Ptr SECP256k1Context)
+
+foreign import capi "secp256k1.h secp256k1_context_destroy"
+  secpContextDestroy :: 
+     Ptr SECP256k1Context
+  -> IO ()
+
+foreign import capi "secp256k1.h value SECP256K1_CONTEXT_SIGN"
+  secpContextSign :: CInt
+
+foreign import capi "secp256k1.h value SECP256K1_CONTEXT_VERIFY"
+  secpContextVerify :: CInt
+
+secpContextSignVerify :: CInt
+secpContextSignVerify = secpContextSign .|. secpContextVerify
+
+data SECP256k1SecKey
+
+data SECP256k1SchnorrSig
+
+data SECP256k1KeyPair
+
+foreign import capi "secp256k1_extrakeys.h secp256k1_keypair_create"
+  secpKeyPairCreate :: 
```

Comment:
are we excluding the prefix because the function is the same for both algorithms?

Reply:
This function is not algorithmically-tied. The `secp256k1_extrakeys.h` header provides general keypair-related manipulation: it is not specific to Schnorr or ECDSA signatures.